### DISCUSSION TITLE
Non integer value for version (e.g. "myformula@hello")

### DISCUSSION THREAD

**Discussion by vilhelmprytz** [Homebrew/discussions#312] [Everyday usage]
Hello!

So I have a formula named `myformula` which installs the latest release of my software. I'd like to add another formula where it installs the latest commit and not the latest release. So I've created another formula named `myformula@edge.rb` where the class is then named `myformulaATedge`. Is this supported? Because when I try to install it locally I get this.

```bash
$ brew install myformula@edge
==> Searching for similarly named formulae...
This similarly named formula was found:
myformula@edge
To install it, run:
  brew install myformula@edge
Error: No available formula or cask with the name "myformula@edge".
```

I find it quite odd because the similarly named formula it finds is exactly the one I am trying to install.

Am I doing something wrong? I couldn't find anything in the documentation hinting towards the versions not supporting characters/letters :slightly_smiling_face:

**Comment by carlocab** [comment]
Have you checked your formula file for syntax errors? I've only ever seen that exact output when there are syntax errors in the formula file.

If you can't find any, try changing the version to a number and try to install it again. If it still gives you the same error then I'm fairly certain you have a syntax error.

**Reply by vilhelmprytz** [reply]
> Have you checked your formula file for syntax errors? I've only ever seen that exact output when there are syntax errors in the formula file.
> 
> If you can't find any, try changing the version to a number and try to install it again. If it still gives you the same error then I'm fairly certain you have a syntax error.

So I changed it to a number and it now returns this.

```bash
$ brew install myformula@2
Error: myformula@2 is a head-only formula
Install with `brew install --HEAD myformula@2`
```

Using `--HEAD` flag seems to work just fine, even though the only thing I did was change the name of the file (and name of the class, `myformulaAT2`).

**Reply by carlocab** [reply]
Well, I think that's your answer: non-number versions (at least for the `@` in a formula name) don't work.

Does `brew install --HEAD` also not work when you've switched back to your previous non-number version?

**Reply by vilhelmprytz** [reply]
Yes, `brew install --HEAD` does not work with a non-number version. Perhaps my problem isn't relevant. I could just have the same `myformula` where `brew install myformula` will install the latest release and `brew install --HEAD myformula` is the latest commit. Anyway, thanks for replying and helping me diagnose the problem, much appreciated! :slightly_smiling_face:
